FINERACT-2421: Add unit tests for ExternalIdFactory

airajena opened the pull request:
## Description
This PR adds comprehensive unit tests for the `ExternalIdFactory` class in the `fineract-core` module.
## Changes
- Added `ExternalIdFactoryTest.java` with 8 test methods covering:
  - `produce(String)` - tests for valid, null, empty, and blank strings
  - `produce(List<String>)` - tests for valid lists, mixed values, null list, and empty list
## Testing
- Tests verify that:
  - Valid strings produce ExternalId with correct value
  - Null/empty/blank strings produce empty ExternalId
  - List factory method correctly converts all values
  - Null list throws NullPointerException as expected
## Related Issue
FINERACT-2421

airajena committed:
FINERACT-2421: Add unit tests for ExternalIdFactory

adamsaghy commented:
```
> Task :fineract-loan:classes
/home/runner/work/fineract/fineract/fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/ExternalIdFactoryTest.java:49: error: reference to produce is ambiguous
```

airajena committed:
FINERACT-2421: Fix ambiguous method reference in ExternalIdFactoryTest

Cast null to String to resolve ambiguous method call between
produce(String) and produce(List<String>) overloads.

airajena commented:
> ```
> > Task :fineract-loan:classes
> /home/runner/work/fineract/fineract/fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/ExternalIdFactoryTest.java:49: error: reference to produce is ambiguous
> ```

Fix pushed

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/command/CalendarCommand.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/command/CalendarCommand.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/serialization/CalendarCommandFromApiJsonDeserializer.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/serialization/CalendarCommandFromApiJsonDeserializer.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/serialization/CalendarCommandFromApiJsonDeserializer.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/serialization/CalendarCommandFromApiJsonDeserializer.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy commented on fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/serialization/CalendarCommandFromApiJsonDeserializer.java:
Not relevant change for testing ExternalIdFactory. Please remove!

adamsaghy reviewed (changes requested):
Please remove not relevant changes!

airajena commented:
> Please remove not relevant changes!

Done

adamsaghy commented:
@airajena Please squash you commits next time!
